client/consensus/src/mocks.rs

```diff
@@ -0,0 +1,816 @@
+        Ok(tree)
+    }
+
+    async fn persisted_validation_data(
+        &self,
+        _hash: PHash,
+        _: ParaId,
+        assumption: OccupiedCoreAssumption,
+    ) -> RelayChainResult<Option<PersistedValidationData>> {
+        let included_persisted = PersistedValidationData {
+            // The parent head-data.
+            parent_head: Default::default(),
+            // The relay-chain block number this is in the context of.
+            relay_parent_number: 0,
+            // The relay-chain block storage root this is in the context of.
+            relay_parent_storage_root: [1u8; 32].into(),
+            // The maximum legal size of a POV block, in bytes.
+            max_pov_size: 5_000_000u32,
+        };
+        let best = self.client.chain_info().best_hash;
+        let header = self
+            .client
+            .header(best)
+            .ok()
+            .flatten()
+            .expect("No header for best");
+
+        let non_included_persisted = PersistedValidationData {
+            parent_head: header.encode().into(),
+            // The relay-chain block number this is in the context of.
+            relay_parent_number: 1,
+            // The relay-chain block storage root this is in the context of.
+            relay_parent_storage_root: Default::default(),
+            // The maximum legal size of a POV block, in bytes.
+            max_pov_size: 5_000_000u32,
+        };
+
+        if assumption == OccupiedCoreAssumption::Included {
+            tracing::info!(target: crate::LOG_TARGET, "included");
+
+            Ok(Some(included_persisted))
+        } else {
+            tracing::info!(target: crate::LOG_TARGET, "non-included");
+
+            Ok(Some(non_included_persisted))
+        }
+    }
+
+    async fn candidate_pending_availability(
+        &self,
+        _: PHash,
+        _: ParaId,
+    ) -> RelayChainResult<Option<CommittedCandidateReceipt>> {
+        unimplemented!("Not needed for test")
+    }
+
+    async fn session_index_for_child(&self, _: PHash) -> RelayChainResult<u32> {
+        Ok(0)
+    }
+
+    async fn import_notification_stream(
+        &self,
+    ) -> RelayChainResult<Pin<Box<dyn Stream<Item = PHeader> + Send>>> {
+        let mut notifications = vec![];
+        for _n in 0..self.block_import_iterations {
+            // It does not matter if we push always the same header, blocks will be created anyway
+            notifications.push(PHeader {
+                parent_hash: Default::default(),
+                number: 1u32,
+                // The state trie merkle root
+                state_root: Default::default(),
+                // The merkle root of the extrinsics.
+                extrinsics_root: Default::default(),
+                // A chain-specific digest of data useful for light clients or referencing auxiliary data.
+                digest: Default::default(),
+            });
+        }
+        Ok(Box::pin(futures::stream::iter(notifications)))
```

Comment:
This function could be made lazy using something like https://docs.rs/futures/latest/futures/stream/fn.unfold.html

This is just to avoid having a vector with 1000 elements in that one test, not sure if worth it.

Reply:
Fixed in [e64f459](https://github.com/moondance-labs/tanssi/pull/662/commits/e64f459bf8462907d9942bc5ec3d8b8713974654) (thank you!)